src/Uno.UWP/Security/Authentication/Web/WebAuthenticationBrokerProvider.iOSmacOS.cs

```diff
@@ -0,0 +1,208 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Security.Authentication.Web;
+using AuthenticationServices;
+using Foundation;
+using Windows.UI.Core;
+using Uno.Extensions;
+#if __MACOS__
+using AppKit;
+#else
+using UIKit;
+using SafariServices;
+#endif
+
+namespace Uno.AuthenticationBroker
+{
+	partial class WebAuthenticationBrokerProvider
+	{
+		private const string asWebAuthenticationSessionErrorDomain = "com.apple.AuthenticationServices.WebAuthenticationSession";
+		private const int asWebAuthenticationSessionErrorCodeCanceledLogin = 1;
+#if __IOS__
+		private const string sfAuthenticationErrorDomain = "com.apple.SafariServices.Authentication";
+		private const int sfAuthenticationErrorCanceledLogin = 1;
+#endif
+		public async Task<WebAuthenticationResult> AuthenticateAsync(
+			WebAuthenticationOptions options,
+			Uri requestUri,
+			Uri callbackUri,
+			CancellationToken ct)
+		{
+			var tcs = new TaskCompletionSource<WebAuthenticationResult>();
+
+			void AuthSessionCallback(NSUrl? callbackUrl, NSError? error)
+			{
+				if (error != null)
+				{
+					if ((error.Domain == asWebAuthenticationSessionErrorDomain &&
+						error.Code == asWebAuthenticationSessionErrorCodeCanceledLogin)
+#if __IOS__
+						|| (error.Domain == sfAuthenticationErrorDomain &&
+							error.Code == sfAuthenticationErrorCanceledLogin)
+#endif
+						)
+					{
+						tcs.SetResult(new WebAuthenticationResult(
+							null,
+							0,
+							WebAuthenticationStatus.UserCancel));
+					}
+					tcs.SetResult(new WebAuthenticationResult(
+						error.ToString(),
+						0,
+						WebAuthenticationStatus.ErrorHttp));
+				}
+				else
+				{
+					tcs.SetResult(new WebAuthenticationResult(
+						callbackUrl?.AbsoluteString,
+						0,
+						WebAuthenticationStatus.Success));
+				}
+			}
+
+			IDisposable? was = default;
+
+			void Cancel()
+			{
+				var w = was;
+				if (w is ASWebAuthenticationSession aswas)
+				{
+					aswas.Cancel();
+				}
+#if __IOS__
+				else if (w is SFAuthenticationSession sfwas)
+				{
+					sfwas.Cancel();
+				}
+#endif
+
+				w?.Dispose();
+				w = null;
```

Comment:
I agree with the robot, unless there's something specific.

Reply:
No it's the result of an incompleted refactoring.  I should reming you the PR is still draft ;-)